apps/heft/src/plugins/TypeScriptPlugin/LinterBase.ts

```diff
@@ -87,8 +85,25 @@ export abstract class LinterBase<TLintResult> {
   public async performLintingAsync(options: IRunLinterOptions): Promise<void> {
     await this.initializeAsync(options.tsProgram);
 
+    const commonDirectory: string = options.tsProgram.getCommonSourceDirectory();
+
+    const relativePaths: Map<string, string> = new Map();
+
+    const fileHash: Hash = createHash('md5');
+    for (const file of options.typeScriptFilenames) {
+      // Need to use relative paths to ensure portability.
+      const relative: string = Path.convertToSlashes(path.relative(commonDirectory, file));
+      relativePaths.set(file, relative);
+      fileHash.update(relative);
+    }
+    const hashSuffix: string = fileHash.digest('base64').replace(/\+/g, '-').replace(/\//g, '_').slice(0, 8);
+
     const tslintConfigVersion: string = this.cacheVersion;
-    const cacheFilePath: string = path.join(this._buildCacheFolderPath, `${this._linterName}.json`);
+    const cacheFilePath: string = path.resolve(
+      this._buildFolderPath,
+      options.tsProgram.getCompilerOptions().outDir || '',
```

Comment:
What does including an empty path segment do?

Reply:
The empty segment will be ignored. So if no outdir is specified, the file will be written to project root.